### PR TITLE
All BO classes should inherit from BObasePage

### PR DESCRIPTION
**Pull request by boubkerbribri**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | All BO classes should inherit from BObasePage
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | npm run smoke-tests

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/15151)

### CODE REVIEW

**Review by SimonGrn — DISMISSED** [review]
Thanks ! I don't know why they were still inheriting from CommonPage, that's much better this way

**Review by matthieu-rolland — DISMISSED** [review]
LGTM

File: tests/puppeteer/.eslintrc.js [hunk 1]
```diff
@@ -16,6 +16,7 @@ module.exports = {
     sourceType: 'module',
   },
   rules: {
-    'no-plusplus': [2, { allowForLoopAfterthoughts: true }]
+    'no-plusplus': [2, { allowForLoopAfterthoughts: true }],
+    "max-len": [2, {"code": 120}]
```
Comment (PierreRambaud) [inline tests/puppeteer/.eslintrc.js]:
```suggestion
    'max-len': [2, {code: 120}]
```

File: tests/puppeteer/pages/commonPage.js [hunk 2]
```diff
@@ -81,11 +81,11 @@ module.exports = class CommonPage {
     switch (parameter) {
       case 'equal':
         await this.page.$eval(selector, el => el.innerText)
-          .then(text => expect(text.replace(/\s+/g, ' ').trim()).to.equal(textToCheckWith));
+          .then(text => global.expect(text.replace(/\s+/g, ' ').trim()).to.equal(textToCheckWith));
```
Comment (PierreRambaud) [inline tests/puppeteer/pages/commonPage.js:84]: why no import expect at the top of the file?
```
import {expect} from 'chai';
```

Comment (boubkerbribri) [reply tests/puppeteer/pages/commonPage.js:84]: if so, we had to import it in each file, so we choose to do it in helper as a global var.

File: tests/puppeteer/.eslintrc.js (hunk 1 above)
Comment (boubkerbribri) [reply tests/puppeteer/.eslintrc.js]: Eslint don't detect errors in eslint config file.
didn't see it, thank you

File: tests/puppeteer/pages/commonPage.js (hunk 2 above)
Comment (PierreRambaud) [reply tests/puppeteer/pages/commonPage.js:84]: Bad programming practice :(

Comment (boubkerbribri) [reply tests/puppeteer/pages/commonPage.js:84]: Ok, good to know, I will do it in an other dedicated PR (to be very simple to review).
Can you tell why it's a bad practice ?

**Review by PierreRambaud — APPROVED** [review]

**Pull request merged**